### PR TITLE
[Rotate] Modify API of rotate(subrange:) to take an arbitrary range expression

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
Just throwing the idea here, maybe would be nice to be able to call this API with any kind of range. 
And since is not a source breaking change, I thought it was worth bring to discussion. 

cc @natecook1000 

### Checklist
- [x] I've added at least one test that validates that my change is working, if appropriate
- [x] I've followed the code style of the rest of the project
- [x] I've read the [Contribution Guidelines](../blob/main/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary

### CODE REVIEW

**Comment by mdznr** [comment]
I think this is a great change. Since you could subscript a collection this way, it makes sense for `rotate` to also let you reference a subrange this way.

File: Sources/Algorithms/Rotate.swift [hunk 1]
```diff
@@ -132,15 +132,17 @@ extension MutableCollection {
...
-  public mutating func rotate(
-    subrange: Range<Index>,
+  public mutating func rotate<Range: RangeExpression>(
+    subrange: Range,
     toStartAt newStart: Index
-  ) -> Index {
-    var m = newStart, s = subrange.lowerBound
-    let e = subrange.upperBound
+  ) -> Index where Range.Bound == Index {
+    let cSubrange = subrange.relative(to: self)
```
Comment (xwu) [inline Sources/Algorithms/Rotate.swift:143]: Question: what does the "c" in `cSubrange` mean?

Comment (LucianoPAlmeida) [reply Sources/Algorithms/Rotate.swift]: I really don't remember sorry, but changed to `rSubrange` for `relative` as it makes more sense I guess.

Comment (xwu) [reply Sources/Algorithms/Rotate.swift]: Perhaps `relativeSubrange` will help everyone understand?

Comment (LucianoPAlmeida) [reply Sources/Algorithms/Rotate.swift]: Just fixed

**Pull request closed without merging**